### PR TITLE
refactor(flex-stacker): linear motion system and motor task

### PR DESCRIPTION
**Pull request by ahiuchingau**
This PR simplifies things a bit in the motor task which helps make implementation of the homing routine a bit easier later.

### CODE REVIEW

File: stm32-modules/include/common/core/linear_motion_system.hpp [hunk 1]
```diff
@@ -5,61 +5,39 @@
 
 namespace lms {
 
-struct BeltConfig {
-    // NOLINTNEXTLINE(misc-non-private-member-variables-in-classes)
-    float pulley_diameter;  // mm
-    [[nodiscard]] constexpr auto get_mm_per_rev() const -> float {
+class BeltConfig {
```
Comment (sfoster1) [inline stm32-modules/include/common/core/linear_motion_system.hpp]: [nit] If everything is going to be public might as well keep it a struct

File: stm32-modules/include/flex-stacker/flex-stacker/motor_task.hpp [hunk 2]
```diff
@@ -299,25 +295,8 @@ class MotorTask {
...
-            case MotorID::MOTOR_L:
-                _l_mech_conf.microstep =
-                    static_cast<float>(pow(2, m.microsteps_power));
-                _l_state.steps_per_mm = _l_mech_conf.get_usteps_per_mm();
-                break;
-            default:
-                break;
-        }
+        motor_state(m.motor_id).lms_config.microstep =
+            static_cast<float>(pow(2, m.microsteps_power));
```
Comment (sfoster1) [inline stm32-modules/include/flex-stacker/flex-stacker/motor_task.hpp]: I think that if you make this `pow(2.f, m.microsteps_power)` it'll remain a float without the static cast

**Review by sfoster1 — APPROVED** [review]
Looks good to me, two nits

File: stm32-modules/include/common/core/linear_motion_system.hpp (hunk 1 above)
Comment (ahiuchingau) [reply stm32-modules/include/common/core/linear_motion_system.hpp]: yes yes

**Review by vegano1 — APPROVED** [review]

**Pull request merged**